test/core/security/credentials_test.cc

```diff
@@ -2451,21 +2502,113 @@ TEST(CredentialsTest,
 }
 
+TEST(
+    CredentialsTest,
+    TestExternalAccountCredsSuccessWithServiceAccountImpersonationAndCustomTokenLifetime) {
+  ExecCtx exec_ctx;
+  Json credential_source = Json::FromString("");
+  TestExternalAccountCredentials::ServiceAccountImpersonation
+      service_account_impersonation;
+  service_account_impersonation.token_lifetime_seconds = 1800;
+  TestExternalAccountCredentials::Options options = {
+      "external_account",    // type;
+      "audience",            // audience;
+      "subject_token_type",  // subject_token_type;
+      "https://foo.com:5555/service_account_impersonation",  // service_account_impersonation_url;
+      service_account_impersonation,      // service_account_impersonation;
+      "https://foo.com:5555/token",       // token_url;
+      "https://foo.com:5555/token_info",  // token_info_url;
+      credential_source,                  // credential_source;
+      "quota_project_id",                 // quota_project_id;
+      "client_id",                        // client_id;
+      "client_secret",                    // client_secret;
+      "",                                 // workforce_pool_user_project;
+  };
+  TestExternalAccountCredentials creds(options, {"scope_1", "scope_2"});
+  // Check security level.
+  GPR_ASSERT(creds.min_security_level() == GRPC_PRIVACY_AND_INTEGRITY);
+  // First request: http put should be called.
+  auto state = RequestMetadataState::NewInstance(
+      absl::OkStatus(),
+      "authorization: Bearer service_account_impersonation_access_token");
+  HttpRequest::SetOverride(
+      httpcli_get_should_not_be_called,
+      external_acc_creds_serv_acc_imp_custom_lifetime_httpcli_post_success,
+      httpcli_put_should_not_be_called);
+  state->RunRequestMetadataTest(&creds, kTestUrlScheme, kTestAuthority,
+                                kTestPath);
+  ExecCtx::Get()->Flush();
+  HttpRequest::SetOverride(nullptr, nullptr, nullptr);
+}
+
+TEST(
+    CredentialsTest,
+    TestExternalAccountCredsFailureWithServiceAccountImpersonationAndInvalidCustomTokenLifetime) {
+  const char* options_string1 =
+      "{\"type\":\"external_account\",\"audience\":\"audience\","
+      "\"subject_token_type\":\"subject_token_type\","
+      "\"service_account_impersonation_url\":\"service_account_impersonation_"
+      "url\",\"service_account_impersonation\":"
+      "{\"token_lifetime_seconds\":599},"
+      "\"token_url\":\"https://foo.com:5555/token\","
+      "\"token_info_url\":\"https://foo.com:5555/token_info\","
+      "\"credential_source\":{\"url\":\"https://foo.com:5555/"
+      "generate_subject_token_format_json\","
+      "\"headers\":{\"Metadata-Flavor\":\"Google\"},"
+      "\"format\":{\"type\":\"json\",\"subject_token_field_name\":\"access_"
+      "token\"}},\"quota_project_id\":\"quota_project_id\","
+      "\"client_id\":\"client_id\",\"client_secret\":\"client_secret\"}";
+  grpc_error_handle error1, error2;
+  auto json1 = JsonParse(options_string1);
```

Comment:
I just realized you are directly using the value of this StatusOr. I think you want to assert json1.ok(). Same for json2.